src/pages/index.js

```diff
@@ -8,62 +8,210 @@ import {
   selectors,
   profileTitleInput,
   profileDescriptionInput,
+  avatarModalFormSelector,
+  editButtonAvatar,
+  avatarSelector,
+  cardDeleteModal,
 } from "../utils/constants.js";
 import UserInfo from "../components/UserInfo.js";
 import FormValidator from "../components/FormValidator.js";
+import PopupWithConfirmation from "../components/PopupWithConfirmation.js";
+import Api from "../utils/Api.js";
+
+const api = new Api({
+  baseUrl: "https://around.nomoreparties.co/v1/group-12",
+  headers: {
+    authorization: "cb447498-fb2c-4c99-9fc5-8ee58bc7fe4c",
+    "Content-Type": "application/json",
+  },
+});
 
 // Popup with Image
 const cardPreview = new PopupWithImage({
   popupSelector: "#preview-image-modal",
 });
 cardPreview.setEventListeners();
 
-// Section / Card
-
+// Section / Card Create Card
 const renderCard = (data) => {
+  const likes = data.likes || [];
   const cardElement = new Card(
     {
       data,
       handleImageClick: (imageData) => {
         cardPreview.open(imageData);
       },
+      handleDeleteClick: () => {
+        deleteModal.setSubmitAction(() => {
+          deleteModal.renderLoading(true);
+          const id = cardElement.getId();
+          api
+            .deleteCard(id)
+            .then(() => {
+              cardElement.deleteButton();
+              deleteModal.close();
+            })
+            .catch((err) => {
+              console.error(err);
+            })
+            .finally(() => {
+              deleteModal.renderLoading(false);
```

Comment:
The button text disappears after first submission

Reply:
Done